### PR TITLE
Fix buffer overrun when parsing from bytes

### LINKED ISSUE #12: jq.JvFromJSONBytes() uses unsafe c string casting.
The byte slice passed to this function is not guaranteed to be null terminated so the cast to a c string could overrun the byte array and include garbage in the c function call.  Here's a simple repro case
```go
package main

import (
	"fmt"
	"github.com/ashb/jqrepl/jq"
)

func main() {
	if _, err := jq.JvFromJSONBytes([]byte("true")); err != nil {
		fmt.Println(err)
	} else {
		fmt.Println("all good")
	}
}
```

If you run it several times, you see the conversion sometimes fails because there are stray bytes after the content:
```
$ go run tmp.go 
all good
$ go run tmp.go 
all good
$ go run tmp.go 
jq: error: Invalid literal at EOF at line 1, column 5 (while parsing 'true')
$ go run tmp.go 
jq: error: Invalid literal at EOF at line 1, column 5 (while parsing 'true')
$ go run tmp.go
```

I'm currently working around this by ensuring the byte slice is terminated, i.e.
```go
	if _, err := jq.JvFromJSONBytes([]byte("true\x00")); err != nil {
```

But this should either be documented or fixed by allocating a separate buffer and copying or converting to string and using `CString` (which of course is copying as well).

### PR DESCRIPTION
**Pull request by ashb**
Closes #12

### CODE REVIEW

File: jq/jv.go [hunk 1]
```diff
@@ -230,7 +230,7 @@ func JvFromJSONString(str string) (*Jv, error) {
 // JvFromJSONBytes takes a utf-8 byte sequence containing JSON and returns the
 // jv representation of it.
 func JvFromJSONBytes(b []byte) (*Jv, error) {
-	jv := C.jv_parse((*C.char)(unsafe.Pointer(&b[0])))
+	jv := C.jv_parse_sized((*C.char)(unsafe.Pointer(&b[0]), C.int(len(b))))
```
Comment (danopia) [inline jq/jv.go:233]:
```suggestion
	jv := C.jv_parse_sized((*C.char)(unsafe.Pointer(&b[0])), C.int(len(b)))
```
I had to move the closing parenthesis in order for this to compile on my machine.

Comment (ashb) [reply jq/jv.go:233]: Thanks!  I hadn't gotten around to looking at it, and didn't even have a go dev env set up so was using a few spare moments and CI to try and fix it!

Comment (danopia) [reply jq/jv.go]: No worries. I just submitted #16 which fixes the broken test that I'm seeing in this CI run.

Your CI doesn't seem to run on PRs from outside contributors, which makes it harder to demonstrate that this fixes the build 😄  and is also how the failing test got merged to begin with

Comment (ashb) [reply jq/jv.go]: Curious, I wonder how I did that. Given it's just CircleCi that wasn't my intent :D

Comment (ashb) [reply jq/jv.go]: Fixed. I had somehow turned that setting off in CircleCI.

**Pull request merged**